src/measurements.ts

```diff
@@ -1,80 +1,186 @@
-    scene.add(this.line);
+    const line = new THREE.Line(geometry, new THREE.LineBasicMaterial({ color: this.color }));
+    this.colors.push(this.color);
+    this.lines.push(line);
+    this.scene.add(line);
+  }
+
+  private convertDistance(distanceInMeters: number, unit: string): number {
+    switch (unit) {
+      case 'm':
+        return distanceInMeters;
+      case 'inch':
+        return distanceInMeters * this.METERS_TO_INCHES;
+      case 'foot':
+        return distanceInMeters * this.METERS_TO_FEET;
+      case 'cm':
+      default:
+        return distanceInMeters * this.METERS_TO_CM;
+    }
   }
 
-  private updateMeasurementDisplay(text: string) {
-    const measurementContainer = document.getElementById("measurementDiv");
-    if (measurementContainer) measurementContainer.innerText = text;
+  private updateMeasurementDisplay() {
+    const measurementsContainer = document.querySelector("#measurements");
+    if (this.lines.length === 0 && this.measurementTable) {
+      this.measurementTable.style.display = "none";
+      return;
+    } else {
+      this.measurementTable.style.display = "block";
+    }
+    const unitDropdown = document.getElementById("unit-dropdown") as HTMLSelectElement;
+    if (!measurementsContainer || !unitDropdown) return;
+
+    const selectedUnit = unitDropdown.value;
+
+    let rows = this.distances.map((distance, index) => {
+      const convertedDistance = this.convertDistance(distance, selectedUnit).toFixed(2);
+      return `
+        <tr>
+          <td style="text-align: center; vertical-align: middle;">
+            <div style="width: 60px; height: 25px; background-color: ${this.colors[index]}; border: 1px solid black;"></div>
+          </td>
+          <td style="text-align: center; vertical-align: middle;">
+            ${convertedDistance} ${selectedUnit}
+          </td>
+          <td style="text-align: center; vertical-align: middle;">
+            <button class="removeLine" data-index="${index}">Remove Line</button>
+          </td>
+        </tr>`;
+    }).join("");
+
+    measurementsContainer.innerHTML = rows;
+    this.addEventListeners();
+  }
+
+  private addEventListeners() {
+    const clearButton = document.getElementById("clearButton");
+    clearButton?.addEventListener("click", () => this.clearAll());
+
+    document.getElementById("unit-dropdown")?.addEventListener("change", () => this.updateMeasurementDisplay());
+
+    document.querySelectorAll(".removeLine").forEach((button) => {
+      const index = parseInt((button as HTMLElement).dataset.index!, 10);
+      button.addEventListener("click", () => this.removeLine(index));
+    });
+  }
+
+  private clearAll() {
+    this.spheres.forEach(({ sphere1, sphere2 }) => {
+      this.scene.remove(sphere1);
+      this.scene.remove(sphere2);
+    });
+    this.lines.forEach(line => this.scene.remove(line));
+
+    this.resetMeasurements();
+    this.updateMeasurementDisplay();
+  }
+
+  private resetMeasurements() {
```

Comment:
```suggestion
  private resetMeasurements() {
```
Remove this method implement it directly where it is calling